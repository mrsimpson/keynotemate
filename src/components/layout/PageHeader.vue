<template>
  <header class="page-header">
    <h1 class="page-title">{{ title }}</h1>
    <div class="page-actions">
      <slot name="actions"></slot>
    </div>
  </header>
</template>

<script setup lang="ts">
defineProps<{
  title: string;
}>();
</script>

<style scoped>
.page-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: var(--spacing-xl);
  gap: var(--spacing-lg);
}

.page-title {
  margin: 0;
  font-size: 2rem;
  color: var(--text);
  flex: 1;
}

.page-actions {
  display: flex;
  gap: var(--spacing-md);
  flex-wrap: wrap;
}

@media (max-width: 600px) {
  .page-header {
    flex-direction: column;
    align-items: stretch;
  }

  .page-title {
    text-align: center;
    margin-bottom: var(--spacing-md);
  }

  .page-actions {
    justify-content: center;
  }
}
</style>