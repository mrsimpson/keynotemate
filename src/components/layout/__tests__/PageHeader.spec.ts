import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import PageHeader from '../PageHeader.vue'

describe('PageHeader Component', () => {
  it('renders basic page header snapshot', () => {
    const wrapper = mount(PageHeader, {
      props: { title: 'Test Title' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders page header with actions snapshot', () => {
    const wrapper = mount(PageHeader, {
      props: { title: 'Test Title' },
      slots: {
        actions: '<button>Action</button>',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders page header with long title snapshot', () => {
    const longTitle = 'A very long title that should be displayed correctly'
    const wrapper = mount(PageHeader, {
      props: { title: longTitle },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
