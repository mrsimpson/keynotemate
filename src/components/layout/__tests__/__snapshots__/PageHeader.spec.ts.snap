// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader Component > renders basic page header snapshot 1`] = `
"<header data-v-075db758="" class="page-header">
  <h1 data-v-075db758="" class="page-title">Test Title</h1>
  <div data-v-075db758="" class="page-actions"></div>
</header>"
`;

exports[`PageHeader Component > renders page header with actions snapshot 1`] = `
"<header data-v-075db758="" class="page-header">
  <h1 data-v-075db758="" class="page-title">Test Title</h1>
  <div data-v-075db758="" class="page-actions"><button>Action</button></div>
</header>"
`;

exports[`PageHeader Component > renders page header with long title snapshot 1`] = `
"<header data-v-075db758="" class="page-header">
  <h1 data-v-075db758="" class="page-title">A very long title that should be displayed correctly</h1>
  <div data-v-075db758="" class="page-actions"></div>
</header>"
`;
